### PR TITLE
[compositorclient:RPI] avoid local variable in the inner block for la…

### PR DESCRIPTION
**Pull request by HaseenaSainul**
…mbda functions

### CODE REVIEW

File: Source/compositorclient/RPI/Implementation.cpp [hunk 1]
```diff
@@ -579,8 +581,8 @@ int Display::Process(const uint32_t data)
         std::function<void(SurfaceImplementation*)> action = nullptr;
 
         if (message.type == KEYBOARD) {
-            const IDisplay::IKeyboard::state state = ((message.keyData.type == KEY_RELEASED)? IDisplay::IKeyboard::released : IDisplay::IKeyboard::pressed);
```
Comment (MFransen69) [inline Source/compositorclient/RPI/Implementation.cpp]: Hi Haseena,

Could you perhaps explain why you are making this change? The state is only used locally in this method, correct?

Comment (HaseenaSainul) [reply Source/compositorclient/RPI/Implementation.cpp]: const IDisplay::IKeyboard::state state is local to this block and it is not accessible outside the block. i.e, here the lambda function action will get invoked outer this block (https://github.com/WebPlatformForEmbedded/Thunder/blob/db2e4b21499b837810d9780d96de045c442da11a/Source/compositorclient/RPI/Implementation.cpp#L620)  and that time this value will be junk. Right now I have kept _key_state as member variable, I can make it as the local to the https://github.com/WebPlatformForEmbedded/Thunder/blob/db2e4b21499b837810d9780d96de045c442da11a/Source/compositorclient/RPI/Implementation.cpp#L574 as well

Comment (MFransen69) [reply Source/compositorclient/RPI/Implementation.cpp]: Yes, sorry missed that it was used in a lambda so not only localy, didn't look any further then my nose is long ;) (Dutch saying for that I did not look long enough :) )
But then whey pass it by reference to the lambda and not value? (in the latter case it still can be kept local) Should the lambda use the value at creation of the lambda or is there a chance it is updateded later and the current value at execution of the lambda must be used?

Comment (HaseenaSainul) [reply Source/compositorclient/RPI/Implementation.cpp]: Here the lambda using only the value at the creation. But if we follow the latter case (pass by value), do we have to use a different function prototype for lambda, right ? In that case, we have touch all other lambda functions created for mouse / touch etc.

**Pull request merged**